api/pkg/apis/v1alpha1/utils/utils.go

```diff
@@ -319,3 +319,10 @@ func jsonPathQuery(obj interface{}, jsonPath string) (interface{}, error) {
 		return result, nil
 	}
 }
+
+func ReplaceSeperator(name string) string {
+	if strings.Contains(name, ":") {
+		name = strings.ReplaceAll(name, ":", "-")
+	}
+	return name
```

Comment:
Shall we throw error if the input names of stage providers don't have a version?

Reply:
You mean to block input names: app-v1? As list stage can return a list of object metadata names, I do think we should not hard set input naming rule.

Reply:
any name without :, user can fill in any string in the field. We need to handle the error case.